[16512] Gateway & Dashboard 5.11.1 RNs.mdx (#1308)
TykTechnologies/tyk-docs: developer-support/release-notes/gateway.mdx

## 5.11 Release Notes 

### 5.11.1 Release Notes 

#### Release 13th February 2025

#### Release Highlights

> In this release, we have fixed a memory leak in the gateway that could occur when using JWT authentication, we have resolved a performance issue with bundle verification that significantly impacted resource consumption when using plugin bundles, and we have fixed some priority CVEs. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v5.11.1) below.

Reviewer: ```suggestion
In this release, we have fixed a memory leak that could occur when using JWT authentication; we have resolved a performance issue with bundle verification that significantly impacted resource consumption when using plugin bundles; and we have fixed some priority CVEs. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v5.11.1) below.
```

#### Breaking Changes

There are no breaking changes in this release.

#### Dependencies 
<a id="dependencies-5.11.1"></a>

##### Compatibility Matrix For Tyk Components

> | Gateway Version | Recommended Releases | Backwards Compatibility |

Reviewer 2: Same here

|--------|-------------------|---- |
| 5.11.1 | MDCB v2.8.7       | MDCB v2.8.7 |
|        | Operator v1.2.0   | Operator v0.17 |
|        | Sync v2.1.5       | Sync v2.1.0 |
|        | Helm Chart v5.0   | Helm all versions |

> |        | Pump v1.13.2      | Pump all versions |

Reviewer 2: ```suggestion
| 5.11.1 | MDCB v2.8.8       | MDCB v2.8.7 |
|        | Operator v1.2.0   | Operator v0.17 |
|        | Sync v2.1.5       | Sync v2.1.0 |
|        | Helm Chart v5.0   | Helm all versions |
|        | Pump v1.13.3      | Pump all versions |
```

Given the potential time difference between your upgrade and the release of this version, we recommend users verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

There are no deprecations in this patch release.

#### Upgrade instructions
<a id="upgrade-5.11.1"></a>

If you are upgrading to 5.11.1, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.11.1)
  - ```bash
    docker pull tykio/tyk-gateway:v5.11.1
    ``` 
- Helm charts
  - [tyk-charts v5.0.0](/developer-support/release-notes/helm-chart#5-0-0-release-notes)

Please note that the Tyk Helm Charts are configured to install the LTS version of Tyk Gateway. You will need to modify them to install v5.11.1.

- [Source code tarball of Tyk Gateway v5.11.1](https://github.com/TykTechnologies/tyk/releases/tag/v5.11.1)

#### Changelog
<a id="Changelog-v5.11.1"></a>

##### Fixed

<AccordionGroup>

> <Accordion title='Fixed Memory Leak in Gateway When Using JWKS URL Cache'>

Reviewer: ```suggestion
<Accordion title='Fixed Memory Leak When Using JWKS URL Cache'>
```

> Resolved a memory leak issue in the Gateway that could occur when APIs used JWT authentication with JWKS URL cache.

Reviewer: ```suggestion
Resolved a memory leak issue that could occur when APIs used JWT authentication with JWKS URL cache.
```

</Accordion>

<Accordion title='Optimized Bundle Verification to Reduce Memory Consumption'>
Fixed a performance issue introduced in v5.8.7 where bundle verification significantly increased CPU and memory consumption, particularly when using multiple APIs with plugin bundles.

> Additionally, introduced a new Gateway configuration option `skip_verify_existing_plugin_bundle` that allows you to skip cryptographic verification when loading signed plugin bundles from disk. When set to `true`, this option reduces the performance overhead for environments with large numbers of APIs using signed bundles, while still maintaining security by validating signatures during the initial bundle download.

Reviewer: ```suggestion
We have introduced a new Gateway configuration option `skip_verify_existing_plugin_bundle` that allows you to skip cryptographic verification when loading signed plugin bundles from disk. When set to `true`, this option reduces the performance overhead for environments with large numbers of APIs using signed bundles, while still maintaining security by validating signatures during the initial bundle download.
```

**Note**: This option only affects signed bundles loaded from disk, unsigned bundles and initial downloads will continue to follow standard verification procedures.
</Accordion>

</AccordionGroup>

##### Security Fixes

<AccordionGroup>

<Accordion title='CVE fixed'>

> Fixed the following CVEs, providing increased protection against security

Reviewer: ```suggestion
Addressed CVEs reported in dependent libraries, providing increased protection against security
```

## 5.11 Release Notes 

### 5.11.1 Release Notes 

#### Release xx 2025

#### Release Highlights

> In this release, we have fixed a memory leak in the gateway that occurred when using JWKS URL authentication. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v5.11.1) below.

Reviewer: ```suggestion
In this release, we have fixed a memory leak in the gateway that could occur when using JWT authentication. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v5.11.1) below.
```

| 5.11.1 | MDCB v2.8.7       | MDCB v2.8.7 |
|        | Operator v1.2.0   | Operator v0.17 |
|        | Sync v2.1.5       | Sync v2.1.0 |
|        | Helm Chart v5.0   | Helm all versions |
|        | Pump v1.13.2      | Pump all versions |

##### 3rd Party Dependencies & Tools

| Third Party Dependency | Tested Versions | Compatible Versions | Comments | 
| ---------------------- | --------------- | ------------------- | -------- | 
| [Go](https://go.dev/dl/)               | 1.24                   |  1.24  | [Go plugins](/api-management/plugins/golang) must be built using Go 1.24 | 
| [Redis](https://redis.io/download/)    | 6.2.x, 7.x, 7.4.x      | 6.2.x, 7.x, 7.4.x      | | 
| [Valkey](https://valkey.io/download/)  | 7.2.x, 8.0.x, 8.1.x    | 7.2.x, 8.0.x, 8.1.x    | | 
| [OpenAPI Specification](https://spec.openapis.org/oas/v3.0.3)| v3.0.x  | v3.0.x | Supported by [Tyk OAS](/api-management/gateway-config-tyk-oas) |

Given the potential time difference between your upgrade and the release of this version, we recommend users verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

> In this release we've deprecated the [policies.allow_explicit_policy_id](/tyk-oss-gateway/configuration#policies-allow-explicit-policy-id) configuration option. This was previously added to allow the use of custom policy IDs, which is now the default behaviour so this option is redundant.

Reviewer: ```suggestion
There are no deprecations in this patch release.
```

#### Upgrade instructions
<a id="upgrade-5.11.1"></a>

If you are upgrading to 5.11.1, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.11.1)
  - ```bash
    docker pull tykio/tyk-gateway:v5.11.1
    ``` 
- Helm charts
  - [tyk-charts v5.0.0](/developer-support/release-notes/helm-chart#5-0-0-release-notes)

Please note that the Tyk Helm Charts are configured to install the LTS version of Tyk Gateway. You will need to modify them to install v5.11.1.

> - [Source code tarball of Tyk Gateway v5.10.1](https://github.com/TykTechnologies/tyk/releases/tag/v5.11.1)

Reviewer: ```suggestion
- [Source code tarball of Tyk Gateway v5.11.1](https://github.com/TykTechnologies/tyk/releases/tag/v5.11.1)
```

#### Changelog
<a id="Changelog-v5.11.1"></a>

##### Fixed

<AccordionGroup>

> <Accordion title='Fixed Memory Leak in Gateway When Using JWKS URL Authentication'>

Reviewer: ```suggestion
<Accordion title='Fixed Memory Leak in Gateway When Using JWKS URL Cache'>
```

> Resolved a memory leak issue in the gateway that occurred when APIs used JWT authentication with JWKS URLs.

Reviewer: ```suggestion
Resolved a memory leak issue in the Gateway that could occur when APIs used JWT authentication with JWKS URL cache.
```
